Add Redis Configuration Analyzer

jeanbrand0n opened the pull request:
### Summary:
This PR adds a new analyzer for Redis configuration files, checking for common misconfigurations like binding to all interfaces, using the default port, and missing log file configurations.

### Technical Details:
*   **New Analyzer:** A new Redis configuration analyzer (`redis_analyzer.py`) has been implemented. This analyzer parses Redis configuration files (`redis.conf`) and identifies potential security misconfigurations.
*   **Misconfiguration Checks:** The analyzer checks for the following:
    *   Binding to all interfaces (0.0.0.0)
    *   Using the default port (6379)
    *   Missing log file configuration
*   **Reporting:** The analyzer generates a report summarizing the identified misconfigurations, including the total number of issues found. The report's priority is set to HIGH if misconfigurations are found and LOW otherwise.
*   **Task Integration:** A new Celery task (`redis_task.py`) has been added to integrate the Redis analyzer into the worker. This task is registered with the appropriate metadata (display name, description, compatible inputs).
*   **Unit Tests:** Comprehensive unit tests (`test_redis_analyzer.py`) have been added to verify the functionality of the Redis analyzer, covering both empty and misconfigured Redis configuration scenarios.
*   **App Update:** The `app.py` file has been updated to include the new `redis_task` in the Celery worker.  This ensures the task is registered and available for execution.

hacktobeer commented on src/analyzers/redis_analyzer.py:
```diff
@@ -0,0 +1,76 @@
+    bind_everywhere_re = re.compile(
+        r'^\s*bind[\s"]*0\.0\.0\.0', re.IGNORECASE | re.MULTILINE)
+    default_port_re = re.compile(r"port\s+6379\b", re.IGNORECASE)
+    missing_logs_re = re.compile(r'^logfile\s+"[^"]+"$', re.MULTILINE)
```
This will trigger on an empty report. So the test for the empty report will fail.

jeanbrand0n replied:
Good catch. 

I fixed this and changed how we report an empty config, which would trigger this.

hacktobeer commented on tests/test_redis_analyzer.py:
```diff
@@ -0,0 +1,62 @@
+        summary = "No issues found in Redis configuration"
+
+        self.assertIsInstance(result, Report)
+        self.assertEqual(result.priority, Priority.LOW)
```
fails as 'logfile' is triggered on empty report

jeanbrand0n replied:
Fixed

hacktobeer reviewed (changes requested):
PTAL

jeanbrand0n commented:
I believe I've addressed these changes now and fixed the tests.

.```
venv ❯ pytest -v
================================================================================================ test session starts ================================================================================================
platform darwin -- Python 3.13.1, pytest-8.3.4, pluggy-1.5.0 -- /Users/chalky/code/openrelik-src/.venv/bin/python3.13
cachedir: .pytest_cache
rootdir: /Users/chalky/code/openrelik-src/openrelik-worker-analyzer-config
configfile: pyproject.toml
collected 8 items                                                                                                                                                                                                   

test_jenkins_analyzer.py::Utils::test_analyze_config PASSED                                                                                                                                                   [ 12%]
test_jenkins_analyzer.py::Utils::test_jenkins_extract_credentials PASSED                                                                                                                                      [ 25%]
test_jenkins_analyzer.py::Utils::test_jenkins_extract_versions PASSED                                                                                                                                         [ 37%]
test_jupyter_analyzer.py::Utils::test_analyse_config PASSED                                                                                                                                                   [ 50%]
test_redis_analyzer.py::Utils::test_redisconfig_empty PASSED                                                                                                                                                  [ 62%]
test_redis_analyzer.py::Utils::test_redisconfig_weak PASSED                                                                                                                                                   [ 75%]
test_sshd_analyzer.py::Utils::test_sshdconfig_empty PASSED                                                                                                                                                    [ 87%]
test_sshd_analyzer.py::Utils::test_sshdconfig_weak PASSED                                                                                                                                                     [100%]

================================================================================================= 8 passed in 0.03s =================================================================================================

openrelik-worker-analyzer-config/tests redis 
```

jeanbrand0n commented:
Closing this PR as a rebase is tricky to comply with the CLA. 

I will open a new one.
